overlord/snapstate/snapstate.go

```diff
@@ -163,6 +167,16 @@ func Update(s *state.State, name, channel string, flags snappy.InstallFlags) (*s
 		return nil, fmt.Errorf("cannot find snap %q", name)
 	}
 
+	if channel == "" { // find a recent channel, usually from current
+		for i := len(snapst.Sequence) - 1; i >= 0; i-- {
```

Comment:
We have Channel in SnapState. That's what I think we should really use here when empty, and when explicitly set we should update it.